gomezmoreno-tp-biblioteca/app.js

```diff
@@ -0,0 +1,253 @@
+    this.cantidad = cantidad;
+    this.disponibles = function(prestamos) {
+        var prestado = 0;
+        for (var librosPrestados of prestamos) {
+            if(librosPrestados.idLibro == this.id) {
+                prestado++;
+            }
+        }
+        return this.cantidad - prestado;
+    } 
+}
+
+function Socio(id_Socio, nombre) {
+    this.id = id_Socio;
+    this.nombre = nombre;
+}
+
+function Prestamo(idPrestamo, idSocio, idLibro, dias) {
+    this.idPrestamo = idPrestamo;
+    this.idSocio = idSocio;
+    this.idLibro = idLibro;
+    this.fechaVencimiento = Date.now() + DIA_EN_MILISEGUNDOS * dias;
+}
+
+var libros = new Array();
+var socios = new Array();
+var prestamos = new Array();
+
+function getNewId() {
+    return Math.random().toString(36).substr(2,9);
+}
+
+function encontrarIndice (coleccion, idABuscar) {
+    function esElId(elemento) {
+        return elemento.id == idABuscar;
+    }
+    return coleccion.findIndex(esElId);
+}
+
+//RQ1.1: Agregar libro a la libreria
+app.post('/libros', function (req, res) {
+    var idLibro = getNewId();
+    var nombreLibro = req.body.titulo;
+    var cantidadLibro = req.body.cantidad;
+    if(Number.isInteger(cantidadLibro) == true && cantidadLibro > 0) {
+        if(nombreLibro != undefined) {
+            libros.push(new Libro(idLibro, nombreLibro, cantidadLibro));
+            res.status(201).json("idLibro: " + idLibro);
+        }
+        else {
+            res.status(400).json("No puede agregarse un libro sin ingresar un titulo");
+        }
+    }
+    else {
+        res.status(400).json("La cantidad debe ser un entero mayor a cero");
+    }
+ })
+
+function libroPrestado(idLibro) {
+    for (var librosPrestamos of prestamos) {
+        if(librosPrestamos.idLibro == idLibro) {
+            return 1;
+        }
+    }
+    return -1;
+}
+
+//RQ1.2: Eliminar libro de la libreria
+app.delete('/libros/:libro', function (req, res) {
+    var eliminarLibro = encontrarIndice(libros, req.params.libro);
+    if(eliminarLibro != -1) {
+        if(libroPrestado(req.params.libro) == -1) {
+            libros.splice(eliminarLibro, 1);
+            res.status(204).json("idLibro: " + eliminarLibro);
+        }
+        else {
+            res.status(400).json("El libro no puede ser eliminado porque está prestado");
+        }
+    }
+    else {
+        res.status(404).json("El id del libro ingresado no existe");
+    }
+})
+
+//RQ1.3: Modificar la cantidad de ejemplares de un libro en la libreria
+app.put('/libros/:libro', function (req, res) {
+    var libroAModificar = encontrarIndice(libros, req.params.libro);
+    var cantidadNueva = req.body.cantidad;
+    if(libroAModificar != -1) {
+        if(Number.isInteger(cantidadNueva) == true && cantidadNueva > 0) {
+            if(cantidadNueva > libros[libroAModificar].disponibles(prestamos)) {
+                libros[libroAModificar].cantidad = cantidadNueva;
+                res.status(204).json("idLibro: " + libroAModificar);
```

Comment:
if you put a response code equal to 204 it won't return a message

Reply:
I changed it to .send